### PR TITLE
Allows non-human bodies to spawn in the morgue at roundstart

### PR DESCRIPTION
**Pull request by ShizCalev**
## About The Pull Request

Adds a configurable probability for the cadavers in the morgue to spawn as nonhuman species.

## Why It's Good For The Game

Downstreams were asking for nonhuman cadavers to be added to the morgue spawns so doctors can at least _attempt_ to help nonhuman species.

## Changelog

:cl: ShizCalev
server: Added a configurable probability for nonhuman species to spawn in the morgue at round start!
qol: Cadavers in the morgue at round start will now spawn in bodybags.
fix: Adminspawned morgue cadavers will now only spawn inside closed morguetrays (the bodies could get stuck inside them!)
fix: Morguetrays are now less likely to end up having multiple cadavers in them.
/:cl:

### CODE REVIEW

File: code/modules/mapping/mapping_helpers.dm [hunk 1]
```diff
@@ -446,28 +446,82 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
...
+	var/reuse_trays = (trays.len < bodycount) //are we going to spawn more trays than bodies?
+
+	var/use_species = CONFIG_GET(flag/morgue_cadaver_disable_nonhumans)
+	var/species_probability = CONFIG_GET(number/morgue_cadaver_other_species_probability)
+	var/override_species = CONFIG_GET(string/morgue_cadaver_override_species)
+	var/list/usable_races
+	if(use_species)
+		usable_races = GLOB.roundstart_races.Copy() //micro optimization
+		usable_races -= SPECIES_ETHEREAL //they revive on death which is bad juju
+		usable_races -= SPECIES_HUMAN
```
[the remark this answers is not in the document]
Comment (ShizCalev) [reply code/modules/mapping/mapping_helpers.dm:484]: Yes! They're set on fire once they're exposed to oxygen. 

To expand on this however, I have added bodybags and clearly labeled the bodybag that there is a plasmaman inside. I felt while opening a morguetray to a flaming body seemed a bit along the lines of a bug, having the bodies inside a bodybag that is clearly labelled that it contains a plasmaman keeps in theme with SS13's gameplay.

File: code/modules/mapping/mapping_helpers.dm [hunk 2]
```diff
@@ -446,28 +446,84 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
...
+					new_human.fully_replace_character_name(new_human.real_name, new_human_species.random_name(new_human.gender, TRUE, TRUE))
+				else
+					stack_trace("failed to spawn cadaver with species ID [species_to_pick]") //if it's invalid they'll just be a human, so no need to worry too much aside from yelling at the server owner lol.
+
+		body_bag.insert(new_human, TRUE)
+		body_bag.close()
+		body_bag.handle_tag("[new_human.real_name][species_to_pick ? " - [capitalize(species_to_pick)]" : " - Human"]")
+		body_bag.forceMove(morgue_tray)
+
+		new_human.death() //here lies the mans, rip in pepperoni.
```
Comment (Cyberboss) [inline code/modules/mapping/mapping_helpers.dm:517]: lol

File: code/game/objects/structures/crates_lockers/closets/bodybag.dm [hunk 3]
```diff
@@ -51,20 +51,19 @@
...
 		to_chat(user, span_notice("You cut the tag off [src]."))
-		name = "body bag"
-		tagged = FALSE
-		update_appearance()
+		handle_tag()
+
+///Handles renaming of the bodybag's examine tag.
+/obj/structure/closet/body_bag/proc/handle_tag(tag_name)
+	name = tag_name ? "[initial(name)] - [tag_name]" : initial(name)
+	tagged = tag_name ? TRUE : FALSE
```
Comment (JohnFulpWillard) [inline code/game/objects/structures/crates_lockers/closets/bodybag.dm:65]:
```suggestion
	tagged = !!tag_name
```

File: code/modules/mapping/mapping_helpers.dm [hunk 4]
```diff
@@ -446,28 +446,84 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
...
+
+	var/reuse_trays = (trays.len < bodycount) //are we going to spawn more trays than bodies?
+
+	var/use_species = CONFIG_GET(flag/morgue_cadaver_disable_nonhumans)
+	var/species_probability = CONFIG_GET(number/morgue_cadaver_other_species_probability)
+	var/override_species = CONFIG_GET(string/morgue_cadaver_override_species)
+	var/list/usable_races
+	if(use_species)
+		usable_races = GLOB.roundstart_races.Copy()
+		usable_races -= SPECIES_ETHEREAL //they revive on death which is bad juju
```
Comment (Kapu1178) [inline code/modules/mapping/mapping_helpers.dm:483]: I really dont like hardcoding stuff like this.

Comment (ShizCalev) [reply code/modules/mapping/mapping_helpers.dm:483]: sadly this is one extremely snowflake where they're literally litches that have their bodies disappear and revive on death, so there's not a way around this.

**Pull request merged**